martha/martha_v3.js

```diff
@@ -174,7 +232,7 @@ function getHttpsUrlParts(url) {
     // The many different ways a DOS/DRS may be "compact", in the order that the should be tried
     const cibRegExps = [
         // Non-W3C CIB DOS/DRS URIs, where the `dg.abcd` appears more than once
-        /(?:dos|drs):\/\/(?<host>dg\.[0-9a-z-]+)(?<separator>:)(?:\k<host>)\/(?<suffix>[^?]*)(?<query>\?(.*))?/i,
+        /(?:dos|drs):\/\/(?<host>dg\.[0-9a-z-]+)(?<separator>:)\k<host>\/(?<suffix>[^?]*)(?<query>\?(.*))?/i,
```

Comment:
FYI IntelliJ suggested this edit.

Reply:
Makes sense. It would throw off some match groups if referenced by position, but we're naming them all so it should be okay.